File: lib/models/labels/email_label.dart
import 'dart:io';

/// Email address label types.
///
/// | Label    | Android | iOS |
/// |----------|:-------:|:---:|
/// | home     | ✔       | ✔   |
/// | iCloud   | ⨯       | ✔   |
/// | mobile   | ✔       | ⨯   |
/// | school   | ⨯       | ✔   |
/// | work     | ✔       | ✔   |
/// | other    | ✔       | ✔   |
/// | custom   | ✔       | ✔   |
enum EmailLabel {
  home,
  iCloud,
  mobile,
  school,
  work,
  other,
  custom;

  static const _android = {
    EmailLabel.home,
    EmailLabel.mobile,
    EmailLabel.work,
    EmailLabel.other,
    EmailLabel.custom,
  };

  static const _apple = {
    EmailLabel.home,
    EmailLabel.iCloud,
    EmailLabel.school,
    EmailLabel.work,
    EmailLabel.other,
    EmailLabel.custom,
  };

  bool get isSupported {
    if (Platform.isAndroid) return _android.contains(this);
    if (Platform.isIOS || Platform.isMacOS) return _apple.contains(this);
    return true;
  }
}

File: ios/flutter_contacts/Sources/flutter_contacts/crud/models/labels/EmailLabel.swift
import Contacts
import Foundation

enum EmailLabel: String, CaseIterable {
    case home
    case iCloud
    case mobile
    case school
    case work
    case other
    case custom

    private static let CN_TO_LABEL: [String: EmailLabel] = [
        CNLabelHome: .home,
        CNLabelWork: .work,
        CNLabelOther: .other,
        CNLabelSchool: .school,
        CNLabelEmailiCloud: .iCloud,
    ]

    private static let LABEL_TO_CN: [EmailLabel: String] = [
        .home: CNLabelHome,
        .work: CNLabelWork,
        .other: CNLabelOther,
        .school: CNLabelSchool,
        .iCloud: CNLabelEmailiCloud,
    ]

    static func fromCN(_ cnLabel: String?) -> Label<EmailLabel> {
        return LabelConverter.fromCN(cnLabel, labelMap: CN_TO_LABEL, defaultLabel: .home)
    }

    func toCNLabel(customLabel: String?) -> String {
        return LabelConverter.toCN(self, customLabel: customLabel, labelMap: EmailLabel.LABEL_TO_CN)
    }
}
